tests/simpleKVBC/TesterReplica/internalCommandsHandler.cpp

```diff
@@ -131,12 +129,47 @@ void InternalCommandsHandler::execute(InternalCommandsHandler::ExecutionRequests
     }
   }
 
-  if (!pre_execute && (merkleUpdates.size() > 0 || verUpdates.size() > 0)) {
+  if (merkleUpdates.size() > 0 || verUpdates.size() > 0) {
     // Write Block accumulated requests
     writeAccumulatedBlock(requests, verUpdates, merkleUpdates);
   }
 }
 
+void InternalCommandsHandler::preExecute(IRequestsHandler::ExecutionRequest &req,
+                                         std::optional<bftEngine::Timestamp> timestamp,
+                                         const std::string &batchCid,
+                                         concordUtils::SpanWrapper &parent_span) {
+  const uint8_t *request_buffer_as_uint8 = reinterpret_cast<const uint8_t *>(req.request);
+  bool readOnly = req.flags & MsgFlag::READ_ONLY_FLAG;
```

Comment:
It still looks that this function does not do all required the things that the 'execute' function does. Why, for example it does not make this check" "if (req.requestSize <= 0)"? Please go through the 'execute' function again and check what else is missing.